### PR TITLE
Fix 91. 'Reset button'. narek

### PR DESCRIPTION
**Pull request by narekacy**
It should be in working condition. But it doesn't work.
You will also need lines 20-24 in `dropdown-group.js`.

### CODE REVIEW

File: client/src/components/dropdown-group.js [hunk 1]
```diff
@@ -7,6 +7,25 @@ import MyDropdown from './my-dropdown';
...
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  shouldComponentUpdate(nextProps, nextState, nextContent) {
+    if (typeof nextProps.categories !== 'undefined' && typeof this.props.categories !== 'undefined') {
+      if (nextProps.categories.length !== this.props.categories.length) {
+        this.setState({ categories: nextProps.categories });
+      }
+    }
+    return this.state.value !== nextState.value;
```
Comment (vitaliimak) [inline client/src/components/dropdown-group.js:26]: It isn't a good idea to use `if` inside another `if`

**Review by vitaliimak — COMMENTED** [review]
Please, don't push `console.log`-s in github. It interferes to see code

Don't forget to use current dev branch. If you don't merge dev into your feature branch, your code can work in feature branch, but it will break after merge and you will need to fix new issue. Or maybe this issue is fixed in current dev branch

Did you try to use another React module instead of `rc-menu`?
You get all events when user open `Events page`, did you try to make filtering only in frontend?

**Comment by narekacy** [comment]
> Please, don't push `console.log`-s in github. It interferes to see code

This is just for testing now.

> Don't forget to use current dev branch. If you don't merge dev into your feature branch, your code can work in feature branch, but it will break after merge and you will need to fix new issue. Or maybe this issue is fixed in current dev branch

I remember this. It will work, don't worry. I will merge later, may be another merges will be before mine.

> Did you try to use another React module instead of `rc-menu`?

`rc-menu` was used by another developer. My goal is to deal with 'Reset' button. On the other hand I agree with you due to React.js components philosophy.

> You get all events when user open `Events page`, did you try to make filtering only in frontend?

No, dude :) This is no either my task.

**Comment by narekacy** [comment]
> > Please, don't push `console.log`-s in github. It interferes to see code
> 
> This is just for testing now.
> 
> > Don't forget to use current dev branch. If you don't merge dev into your feature branch, your code can work in feature branch, but it will break after merge and you will need to fix new issue. Or maybe this issue is fixed in current dev branch
> > I remember this. It will work, don't worry. I will merge later, may be another merges will be before mine.
> 
> > Did you try to use another React module instead of `rc-menu`?
> > `rc-menu` was used by another developer. My goal is to deal with 'Reset' button. On the other hand I agree with you due to React.js components philosophy.
> 
> > You get all events when user open `Events page`, did you try to make filtering only in frontend?
> > No, dude :) This is no either my task.

File: client/src/components/dropdown-group.js (hunk 1 above)
Comment (narekacy) [reply client/src/components/dropdown-group.js:26]: This is just for testing. If this works, of course I will make code better then. Thanks

**Pull request closed without merging**